bytestring_splitter/__init__.py

```diff
@@ -101,19 +109,27 @@ def __call__(self, splittable, return_remainder=False, msgpack_remainder=False):
             else:
                 value = message
 
-            try:
-                processed_objects[message_name] = value
-            except TypeError:
-                processed_objects.append(value)
             cursor = expected_end_of_object_bytes
 
+            if single:
+                _remainder = len(splittable[cursor:])
+                if _remainder:
+                    raise ValueError(f"The bytes don't represent a single {message_class}; there are {_remainder} too many.")  # TODO
+                return value
+            else:
+                try:  # TODO: Make this more agnostic toward the collection type.
```

Comment:
Because it might not throw a `TypeError`?

Reply:
I'm not sure I fully understand the question - maybe you mean that it might not throw `TypeError` but also not implement `append`?

That's certainly a possibility if the underlying logic here is meant to be understood as container agnostic.

Maybe it's fine - the API can be understood such that the collection needs to either implement `__setitem__` or `append`.  If that's so, I think a tiny dose of magic dust to allow a `set` to be used in order, and then this TODO can be wiped.